### ISSUE TITLE
Encoding::InvalidByteSequenceError

### ISSUE THREAD

**Issue by theaamanda** [moumar/ruby-mp3info#66]
I'm fairly new to ruby and wanted to use your awesome mp3info-program! Thank you so much for giving it out to the public!

Its shooting an error message though (it works fine for numerous of tracks - so I guess the error is on my side rather than on your side, but you might be able to help?!) and I would be happy when you could help:

`/Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:491:in`encode!': incomplete "\x00" on UTF-16LE (Encoding::InvalidByteSequenceError)
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:491:in `decode_tag'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:563:in`add_value_to_tag2'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:529:in `block in read_id3v2_3_frames'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:515:in`loop'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:515:in `read_id3v2_3_frames'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:374:in`from_io'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:554:in `parse_tags'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:259:in`reload'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:230:in `initialize'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:300:in`new'
    from /Library/Ruby/Gems/2.0.0/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:300:in `open'`

Thank you

**Comment by alexey** [comment]
Same  issue here, trying to find out why ..

ruby-mp3info-0.8.9
rails 4.2.0
ruby 2.1.1p76

**Comment by alexey** [comment]
Trace:

```
Encoding::InvalidByteSequenceError: incomplete "\x00" on UTF-16LE
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:491:in `encode!'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:491:in `decode_tag'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:563:in `add_value_to_tag2'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:529:in `block in read_id3v2_3_frames'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:515:in `loop'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:515:in `read_id3v2_3_frames'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info/id3v2.rb:374:in `from_io'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:554:in `parse_tags'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:259:in `reload'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:230:in `initialize'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:300:in `new'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/ruby-mp3info-0.8.9/lib/mp3info.rb:300:in `open'
	from (irb):5
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/railties-4.2.0/lib/rails/commands/console.rb:110:in `start'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/railties-4.2.0/lib/rails/commands/console.rb:9:in `start'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/railties-4.2.0/lib/rails/commands/commands_tasks.rb:68:in `console'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/railties-4.2.0/lib/rails/commands/commands_tasks.rb:39:in `run_command!'
	from /Users/alexey/.rvm/gems/ruby-2.1.1@echo-2/gems/railties-4.2.0/lib/rails/commands.rb:17:in `<top (required)>'
	from bin/rails:4:in `require'
```

**Comment by alexey** [comment]
@theaamanda if you have urgently something to work you can use my fork/commit:
https://github.com/alexey/ruby-mp3info/commit/734089bbec9c527c4a1d79daa3b43373c52b08c3

I had to apply very fast solution so i just removed 'comments' tag reading from mp3.
Please don't use it if you rely on 'comments' or anything that may broke your code.

**Comment by hugodemiglio** [comment]
@alexey In my case, all fields of some song are returning the problem. Even using your fork.

If you want to download this song: (free file, downloaded directly from artist website)
http://hiperload.com/s/um6j46r8e

Trace:
```
Encoding::InvalidByteSequenceError: incomplete "\x00" on UTF-16LE
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info/id3v2.rb:491:in `encode!'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info/id3v2.rb:491:in `decode_tag'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info/id3v2.rb:563:in `add_value_to_tag2'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info/id3v2.rb:529:in `block in read_id3v2_3_frames'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info/id3v2.rb:515:in `loop'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info/id3v2.rb:515:in `read_id3v2_3_frames'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info/id3v2.rb:374:in `from_io'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info.rb:554:in `parse_tags'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info.rb:259:in `reload'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info.rb:230:in `initialize'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info.rb:300:in `new'
	from /Users/hugodemiglio/.rvm/gems/ruby-2.1.2@hiperload/bundler/gems/ruby-mp3info-734089bbec9c/lib/mp3info.rb:300:in `open'
```

**Comment by moumar** [comment]
hello everybody,
this bug is (finally) fixed in v0.8.10
